processing/src/main/java/org/apache/druid/query/topn/TopNQueryQueryToolChest.java

```diff
@@ -558,7 +558,16 @@ public Optional<Sequence<FrameSignaturePair>> resultsAsFrames(
       boolean useNestedForUnknownTypes
   )
   {
-    final RowSignature rowSignature = resultArraySignature(query);
+    final RowSignature rowSignature =
```

Comment:
similarily to `TS`: - shouldn't this be `TopNQuery#getResultRowSignature` ?
please also update `resultArraySignature` to use that method so we are not duplicating logic